scripts/azure_cleanup.py

```diff
@@ -36,68 +36,32 @@ def azure_cleanup(nic_template, pip_template, days_old):
-                            removed_stacks.append(stack)
-                            mgmt.delete_stack(stack)
-
-                    logger.info('following stacks were removed:')
-                    for stack in removed_stacks:
-                        logger.info([stack])
-                else:
-                    logger.info("No stacks older than '{}' days were found".format(
-                        days_old))
-
-                """
-                Blob removal section
-                """
-                # TODO: update it later to use different subscriptions and resource groups
-                logger.info("Removing 'bootdiagnostics-test*' containers")
-                bootdiag_list = []
-                for container in mgmt.container_client.list_containers():
-                    if container.name.startswith('bootdiagnostics-test'):
-                        bootdiag_list.append(container.name)
-                        mgmt.container_client.delete_container(
-                            container_name=container.name)
-
-                logger.info('following disks were removed:')
-                for disk in bootdiag_list:
-                    logger.info([disk])
-
-                logger.info("Removing unused blobs and disks")
-                removed_disks = mgmt.remove_unused_blobs()
-                if len(removed_disks['Managed']) > 0:
-                    logger.info('Managed disks:')
-                    logger.info(removed_disks['Managed'])
-
-                if len(removed_disks['Unmanaged']) > 0:
-                    logger.info('Unmanaged blobs:')
-                    logger.info(removed_disks['Unmanaged'])
+                    for resource_group in mgmt.list_resource_groups():
+                        mgmt.logger.info('Checking "{}" resource group:'.format(resource_group))
+
+                        # removing stale nics
+                        mgmt.remove_nics_by_search(nic_template, resource_group)
+
+                        # removing public ips
+                        mgmt.remove_pips_by_search(pip_template, resource_group)
+
+                        # removing stale stacks
+                        stack_list = mgmt.list_stack(resource_group=resource_group,
+                                                     days_old=days_old)
+                        if stack_list:
+                            removed_stacks = []
+                            for stack in stack_list:
+                                if mgmt.is_stack_empty(stack, resource_group=resource_group):
+                                    removed_stacks.append(stack)
+                                    mgmt.delete_stack(stack, resource_group)
+
+                            if not removed_stacks:
+                                logger.info("No empty stacks older '{}' days were found".format(
+                                    days_old))
+                        mgmt.remove_unused_blobs(resource_group)
```

Comment:
**Question**

Why remove all the logging around this call that was there before?

What about the delete_container call from the removed block, I don't see it called anywhere anymore.

Reply:
delete_container was used for removal of bootdiag* containers. we had separate method for that. and I added it's usage into remove_unused_blobs method